mitosheet/src/jupyter/extensionUtils.tsx

```diff
@@ -29,12 +29,82 @@ export function getCellAtIndex(cells: CellList | undefined, index: number): ICel
     return cell
 }
 
+export function getCellIndexByID(cells: CellList | undefined, cellID: string | undefined): number | undefined {
+    if (cells == undefined || cellID == undefined) {
+        return undefined
+    }
+
+    const cellIndex = Array.from(cells).findIndex(cell => cell.id === cellID)
+    return cellIndex === -1 ? undefined : cellIndex;
+}
+
+export function getCellIndexByExecutionCount(cells: CellList | undefined, executionCount: number | undefined): number | undefined {
+    if (cells == undefined || executionCount == undefined) {
+        return undefined;
+    }
+
+    // In order to get the cell index, we need to iterate over the cells and call the `get` method
+    // to see the cells in order. Otherwise, the cells are returned in a random order.
+    for (let i = 0; i < cells.length; i++) {
+        const cell = cells.get(i)
+        if (cell.type === 'code') {
+            const nonTypeSafeCell = cell as any
+            const executionCountEntry = nonTypeSafeCell.sharedModel.ymodel._map.get('execution_count')['content']['arr'][0];
```

Comment:
Try to find a better way of doing this that doesn't access private variables. Is there a getter for this?